test/HealthChecks.Nats.Tests/Functional/NatsHealthCheckTests.cs

```diff
@@ -1,125 +1,105 @@
 using System.Net;
-using System.Text.Json;
-using System.Text.Json.Serialization;
-using HealthChecks.UI.Client;
-using HealthChecks.UI.Core;
+using NATS.Client.Core;
 using static HealthChecks.Nats.Tests.Defines;
 
 namespace HealthChecks.Nats.Tests.Functional;
 
 public class nats_healthcheck_should
 {
     [Fact]
-    public Task be_healthy_if_nats_is_available_locally() =>
-        FactAsync(
-            setup => setup.Url = DefaultLocalConnectionString,
-            async response => response.StatusCode.ShouldBe(HttpStatusCode.OK, await response.Content.ReadAsStringAsync()));
-
-    [Fact]
-    public Task be_healthy_for_official_demo_instance() =>
-        FactAsync(
-            setup => setup.Url = DemoConnectionString,
-            async response => response.StatusCode.ShouldBe(HttpStatusCode.OK, await response.Content.ReadAsStringAsync()));
-
-    [Fact]
-    public Task be_healthy_if_nats_is_available_and_has_custom_name() =>
-        FactAsync(
-            setup => setup.Url = DefaultLocalConnectionString,
-            async response => response.StatusCode.ShouldBe(HttpStatusCode.OK, await response.Content.ReadAsStringAsync()),
-            name: "Demo");
-
-    [Fact]
-    public Task be_unhealthy_if_nats_endpoint_does_not_exist_or_is_offline() =>
-        FactAsync(
-            setup => setup.Url = ConnectionStringDoesNotExistOrStopped,
-            async response =>
+    public async Task be_healthy_when_nats_is_available_using_client_factory()
+    {
+        var webHostBuilder = new WebHostBuilder()
+            .ConfigureServices(services =>
             {
-                response.StatusCode.ShouldBe(HttpStatusCode.ServiceUnavailable);
-                var content = await response.Content.ReadAsStringAsync();
-                var report = JsonSerializer.Deserialize<UIHealthReport>(content, new JsonSerializerOptions { PropertyNameCaseInsensitive = true, Converters = { new JsonStringEnumConverter() } })!;
-                report.Status.ShouldBe(UIHealthStatus.Unhealthy);
-                report.Entries["nats"].Exception.ShouldBe("Failed to connect");
-
-            });
-
-    [Fact]
-    public Task be_unhealthy_if_nats_endpoint_is_bogus() =>
-        FactAsync(
-            setup => setup.Url = "bogus",
-            async response =>
+                services
+                    .AddHealthChecks()
+                    .AddNats(
+                        clientFactory: sp =>
```

Comment:
nit: some of our users may be searching for the samples in our test project. This code is valid, but it registers a factory that is going to create a new connection instance every time a health check is going to be invoked (which may lead to resource leaks etc).

Would you have some time to use NATS.Extensions.Microsoft.DependencyInjection instead and just call `AddNatsClient` from that package?

Here I've found some samples (in a test project ;)): https://github.com/nats-io/nats.net/blob/3f919ddcd55ea0f765b4ce9b4137e2a58c589683/tests/NATS.Extensions.Microsoft.DependencyInjection.Tests/NatsHostingExtensionsTests.cs#L73

Reply:
I did not use `AddNatsClient` in this test because I wanted to test the client factory.
But you are right. It may lead to resource leaks. So, I kept the client factory but changed it to return the created client, not run a new one.